src/java/org/apache/cassandra/auth/IInternodeAuthenticator.java

```diff
@@ -35,10 +36,51 @@
      */
     boolean authenticate(InetAddress remoteAddress, int remotePort);
 
+    /**
+     * Decides whether a peer is allowed to connect to this node.
+     * If this method returns false, the socket will be immediately closed.
+     *
+     * @param certificates client certificates
+     * @return true if the connection should be accepted, false otherwise.
+     */
+    default boolean authenticate(Certificate[] certificates)
+    {
+        return false;
+    }
+
+
+    /**
+     * Decides whether a peer is allowed to connect to this node.
+     * If this method returns false, the socket will be immediately closed.
+     *
+     * @param remoteAddress ip address of the connecting node.
+     * @param remorePort port of the connecting node.
+     * @param certificates client certificates
+     * @return true if the connection should be accepted, false otherwise.
+     */
+    default boolean authenticate(InetAddress remoteAddress, int remorePort, Certificate[] certificates)
```

Comment:
The other piece of information that could be useful is whether this is an inbound or outbound internode connection being authenticated. Organizations may have different standards for the two types of collection and/or create different audit messages depending on the connection. 

Maybe an `enum` argument for connections direction `INTERNODE_INBOUND`, `INTERNODE_OUTBOUND` ?